homeassistant/components/alexa/capabilities.py

```diff
@@ -715,4 +780,267 @@ def configuration(self):
+
+        return None
+
+    def configuration(self):
+        """Return configuration."""
+        configuration = self.serialize_mode_resources()
+        if configuration is not None:
+            return configuration
+        return None
+
+    def capability_resources(self):
+        """Return capabilityResources object."""
+        if self.instance() == fan.ATTR_DIRECTION:
+            return [{"type": Catalog.LABEL_ASSET, "value": Catalog.SETTING_DIRECTION}]
+
+        return None
+
+    def mode_resources(self):
+        """Return modeResources object."""
+        mode_resources = None
+        if self.instance() == fan.ATTR_DIRECTION:
+            mode_resources = {
+                "ordered": False,
+                "resources": [
+                    {
+                        "value": f"{fan.ATTR_DIRECTION}.{fan.DIRECTION_FORWARD}",
+                        "friendly_names": [
+                            {"type": Catalog.LABEL_TEXT, "value": fan.DIRECTION_FORWARD}
+                        ],
+                    },
+                    {
+                        "value": f"{fan.ATTR_DIRECTION}.{fan.DIRECTION_REVERSE}",
+                        "friendly_names": [
+                            {"type": Catalog.LABEL_TEXT, "value": fan.DIRECTION_REVERSE}
+                        ],
+                    },
+                ],
+            }
+
+        return mode_resources
+
+    def serialize_mode_resources(self):
+        """Return ModeResources, friendlyNames serialized for an API response."""
+        mode_resources = []
+        resources = self.mode_resources()
+        ordered = resources["ordered"]
+        for resource in resources["resources"]:
+            mode_value = resource["value"]
+            friendly_names = resource["friendly_names"]
+            result = {
+                "value": mode_value,
+                "modeResources": {
+                    "friendlyNames": self.serialize_friendly_names(friendly_names)
+                },
+            }
+            mode_resources.append(result)
+
+        return {"ordered": ordered, "supportedModes": mode_resources}
+
+
+class AlexaRangeController(AlexaCapibility):
+    """Implements Alexa.RangeController.
+
+    https://developer.amazon.com/docs/device-apis/alexa-rangecontroller.html
+    """
+
+    def __init__(self, entity, instance, non_controllable=False):
+        """Initialize the entity."""
+        super().__init__(entity)
+        self.instance = lambda: instance
+        self.properties_non_controllable = lambda: non_controllable
+
+    def name(self):
+        """Return the Alexa API name of this interface."""
+        return "Alexa.RangeController"
+
+    def properties_supported(self):
+        """Return what properties this entity supports."""
+        return [{"name": "rangeValue"}]
+
+    def properties_proactively_reported(self):
+        """Return True if properties asynchronously reported."""
+        return True
```

Comment:
This should be `alexa_config.should_report_state`

Reply:
I can address this is another PR. It would require the change to all the controllers.